source/memory/memory.go

```diff
@@ -113,6 +129,34 @@ func (s *memorySource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
+// detectSwap detects Swap node information
+func detectSwap() (map[string]string, error) {
+	procBasePath := hostpath.ProcDir.Path("swaps")
+	file, err := os.Open(procBasePath)
+	defer func() {
```

Comment:
Only defer close if the open succeeded -> move this under the error checking below?

Reply:
not sure this is needed with the simplication.